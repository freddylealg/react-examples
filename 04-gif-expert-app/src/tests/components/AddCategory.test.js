import React from 'react';
import { shallow } from 'enzyme'
import '@testing-library/jest-dom';
import toJson from 'enzyme-to-json';
import AddCategory from '../../components/AddCategory';

describe('Pruebas de AddCategory', () => {
    
    const setCategory = jest.fn();
    let wrapper = shallow( <AddCategory setCategories={ setCategory } /> )

    beforeEach( () => { 
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategories={ setCategory } /> );
    });

    test('Debe de mostrar correctamente el componente', () => {
        expect( toJson(wrapper) ).toMatchSnapshot();
    });

    test('Debe de cambiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'Hola Mundo';
        input.simulate('change', {target: {value} });
    });
    
    test('Prueba del submit del formulario, no debe de llamar el formulario', () => {
        wrapper.find('form').simulate('submit', { preventDefault(){} });
        expect( setCategory ).not.toHaveBeenCalled()
    });

    test('Debe de llamar setCategories y limpiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'Vegeta';
        
        input.simulate('change', {target: {value} });
        
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect( setCategory ).toHaveBeenCalled();
        expect( setCategory ).toHaveBeenCalledTimes(1);
        expect( setCategory ).toHaveBeenCalledWith( expect.any(Function) )
        
        expect( input.prop('value') ).toBe( '' );
    })
    

})
