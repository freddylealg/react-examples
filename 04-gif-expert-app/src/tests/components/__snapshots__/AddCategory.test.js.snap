// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de AddCategory Debe de mostrar correctamente el componente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="AÃ±adir categoria"
    type="text"
    value=""
  />
</form>
`;
